### ISSUE TITLE
DREF Account page enhacement

### ISSUE THREAD

**Issue by anamariaescobar** [IFRCGo/go-frontend#2749]
Hello everyone!

Please find the wirreframes for the changes on the DREF account page [here:](https://www.figma.com/proto/u0XifdJwTEx165x3e0eoP9/IFRC-GO_UI-Prototype?type=design&node-id=7006-215581&scaling=min-zoom&page-id=0%3A4957&starting-point-node-id=0%3A4963)

Basically what we want to do is re-organize all the documents related to one DREF (Application, Ops Update, Final report) and group them so we can see them in a timeline like visualization. We are also replacing the word "Publish" by "Approved", the functionality will be the same. We will introduce a new page for the "Completed DREF Operations", an operation can be considered Completed once the final Report is approved.
We also want to bring the sharing functionality on the account page. The allocation form is an export that hasn't been implemented yet, if we can hide this button for now that would be great!

@udaynwa @frozenhelium @tovari @mariam-yu

**Comment by udaynwa** [comment]
@anamariaescobar , thank you for creating the ticket and adding the wireframes. I have added a couple of questions/comments in the wireframe and CC-ed you. 

cc @frozenhelium @tovari

**Comment by Shhhhhubh** [comment]
I tested the DREF Account page and found the following issues @puranban @udaynwa 

- [ ] Status is shown as 1 or 0 (in numbers) instead of submitted, in progress or approved
- [ ] In the filters section - "select type" and "Date range" are missing
- [ ] "View previous DREF operation" is "View Completed DREF operation"
- [ ] The created on date format is "DD-MM-YYYY" instead of "YYYY-MM-DD"
- [ ] In Final report the link bullets (The ones that should be seen after the drop down button is clicked i.e, the red bullets) are missing
- [ ] "Type of DREF" is shown as "Type of Display"
- [ ] When the last page is selected and the filters are changed the pagination sticks to the last page and shows no data
- [ ] Instead of kebab menu buttons we have meatballs menu buttons
- [ ] when the kebab menu button is clicked the view option is missing
- [ ] When the "approved" button is clicked the button should change to "Allocation form" link and this is not happening ( the button disappears this should happen only for final reports)
- [ ] In Previous DREF operations "Type of display" is being shown instead of "Submitted/Approved on"  (a completely different column )
- [ ] In Previous DREF operations the kebab menu buttons is missing
- [ ] In Previous DREF operations the "Approved" button is shown which should be "Allocation Form" also clicking the "Approved" button does nothing 
- [ ] In Previous DREF operation's filter section "Date range" and "select type" is missing
- [ ] Inside the kebab menu the "View" option is missing

![image](https://github.com/IFRCGo/go-frontend/assets/117332528/2a16fad8-c9cc-4c2c-b813-c0dc32d7e826)

**Comment by anamariaescobar** [comment]
@Shhhhhubh thanks for the detail test!! just one comment from my side, for the two following "issues":

- [ ] When the "approved" button is clicked the button should change to "Allocation form" link and this is not happening ( the button disappears this should happen only for final reports)
- [ ] In Previous DREF operations the "Approved" button is shown which should be "Allocation Form" also clicking the "Approved" button does nothing

this might been related with the fact that we haven't developed the "Allocation form", maybe that is why on the first issue this button is being hidden?
So, the final allocation form has been approved from the DREF team, this will be the next step for us to develop after we finish the account page, so we have 2 options, we either hide that button or we just show the the Allocation form button and we just leave it without action until we develop the allocation form. From my side there is no issue leaving the button with no action  

@udaynwa @puranban

**Comment by udaynwa** [comment]
@anamariaescobar , for `submitted/approved on` date column, do we pull the date from the form or do we show the date in which the form is `approved`?

If we are pulling from the form, there is no `Date of Approval` field in the Final Report.

![image](https://github.com/IFRCGo/go-frontend/assets/72339626/c82332fa-9be8-4b4b-926a-2e2b545bdeaa)

cc @frozenhelium

**Comment by udaynwa** [comment]
@Shhhhhubh , for this feedback `The created on date format is "DD-MM-YYYY" instead of "YYYY-MM-DD"`, the standard date format of GO is 'DD-MM-YYYY'. So, I am ticking this feedback as done.

cc @anamariaescobar

**Comment by anamariaescobar** [comment]
Can we get the submitted/approved date from the form for application and Ops update and for the final report the date is approved? @udaynwa @Shhhhhubh

**Comment by mariam-yu** [comment]
> @Shhhhhubh , for this feedback `The created on date format is "DD-MM-YYYY" instead of "YYYY-MM-DD"`, the standard date format of GO is 'DD-MM-YYYY'. So, I am ticking this feedback as done.
> 
> cc @anamariaescobar

Hey guys, for the date we put year first to make it neutral world-wide, as some countries (e.g. US) puts the month first. @tovari shall we consider making this update across?

**Comment by udaynwa** [comment]
> Can we get the submitted/approved date from the form for application and Ops update and for the final report the date is approved? @udaynwa @Shhhhhubh

@anamariaescobar , that is doable. We will implement it accordingly. One quick question would be, can anyone add `Date of Approval`? Or is there some sort of permission tied up with it?

**Comment by anamariaescobar** [comment]
Helloo!! We just finalized reviewing the changes to the account page in staging, a couple of comments from our side:

- [ ] Can we changed the "Approved" button to be moved to the 3 points dropdown and instead have and "Edit" button in it's place? we realized the primary action most of the users look it he account page is editing the form. Once the document is "approved" can the "Edit" button change to "Export"?
- [ ] When "Approved" the sharing option should appear on the dropdown
- [ ] The option to move to a new stage (Ops Update or Final report) should only appear when the previous one is "Approved"
- [ ] When I give the regional permissions to the regional focal points, the focal points should be able to see every Application, Ops Update, etc (even if they are not on the "share" field).
- [ ] There are no permissions linked to the "Date of approval"

@udaynwa @Shhhhhubh

**Comment by udaynwa** [comment]
@anamariaescobar , a quick note, 

For
> When I give the regional permissions to the regional focal points, the focal points should be able to see every Application, Ops Update, etc (even if they are not on the "share" field).

We will be creating a new user group similar to DREF Regional coordinators, but this regional focal points will have access to view (edit?) but not approve, right?

If yes, let's track this in a separate ticket. It is not a very big task, but it will be better if we have a separate ticket for this.

cc @frozenhelium

**Comment by udaynwa** [comment]
@anamariaescobar , the latest works in DREF Account page enhancement have been pushed to Staging. The tasks in this Github ticket have been updated as well. 

cc @tovari @frozenhelium

**Comment by anamariaescobar** [comment]
Thanks! A few comments from our side:

- [ ] The option to move to a new stage (Ops Update or Final report) should only appear when the previous one is "Approved" --> So, if an "Application" is approved, "Ops Update" and "Final Report" should show on the dropdown of the application, if an "Ops Update" is approved then "Final Report" should show on the "Ops Update" dropdown, not on the "Application" one.
- [ ] When the "Final Report" is approved, it should move to the "Completed Operations" and not show on the "Active Operations" anymore.
- [ ] On the column "Status" I still se some empty and "Completed" statuses, we ahould only have "In progress" when they are created and "Approved" once the approved button has been clicked.

The rest looks fantastic!

@udaynwa @frozenhelium @tovari

**Comment by udaynwa** [comment]
Hi there, @anamariaescobar . All the comments have been resolved and pushed to Staging. Please feel free to do full-flow testing and verification of the DREF Account page enhancement on Staging. 

cc @tovari @frozenhelium

**Comment by anamariaescobar** [comment]
Everything looks good from my side!! We are ready to deploy this to Production.

@udaynwa @frozenhelium @tovari
